### PR TITLE
Update logging configuration for better debug and error tracking

### LINKED ISSUE #3192: Ability to log in a file
It would be nice to have logs in a file, because right now log is managed just from console.

### PR DESCRIPTION
**Pull request by RafaelJohn9**
Fix #3192 
- Added rotating file handlers for Django and Pontoon loggers
- Console logging now uses verbose format in DEBUG mode
- Improved log formatting for clarity and consistency

- added *.log  in .gitignore to prevent log files from being pushed

### CODE REVIEW

File: ruff.toml
Comment (flodolo) [inline ruff.toml:1]: Please revert this change in permission (not sure if you meant to commit something else, given the [commit subject](https://github.com/mozilla/pontoon/pull/3260/commits/e908898d89584eec57c21d38b44626db61fd3b4a)).

Comment (RafaelJohn9) [reply ruff.toml:1]: oh yea  sorry about that, I was trying to fix the ruff format style in /pontoon/settings/base.py
but sadly I have confilicting views 

github actions pylint ruff says my ruff format has conflicts  on the file

but locally I have this
image

Comment (flodolo) [reply ruff.toml:1]: Are you running ruff through make? If not, what's the version of ruff?

I checked locally with 0.4.7, and I see the same error as the CI

```
diff --git a/pontoon/settings/base.py b/pontoon/settings/base.py
index 03c629d8..fd419406 100644
--- a/pontoon/settings/base.py
+++ b/pontoon/settings/base.py
@@ -764,14 +764,14 @@ LOGGING = {
         },
         "django_file": {
             "class": "logging.handlers.RotatingFileHandler",
-            "filename": os.path.join(path('django_debug.log')),
+            "filename": os.path.join(path("django_debug.log")),
             "maxBytes": 1024 * 1024 * 2,  # 2 MB
             "backupCount": 3,
             "formatter": "verbose" if DEBUG else "simple",
         },
         "pontoon_file": {
             "class": "logging.handlers.RotatingFileHandler",
-            "filename": os.path.join(path('pontoon_debug.log')),
+            "filename": os.path.join(path("pontoon_debug.log")),
             "maxBytes": 1024 * 1024 * 2,  # 2 MB
             "backupCount": 3,
             "formatter": "verbose" if DEBUG else "simple",
```

Comment (RafaelJohn9) [reply ruff.toml:1]: and my work tree is clean

Comment (RafaelJohn9) [reply ruff.toml:1]: No am not running it through make, my ruff version is 0.4.6

File: pontoon/settings/base.py [hunk 1]
```diff
@@ -753,18 +753,48 @@ def _default_from_email():
 )
 
 # Logging
+
```
Comment (mathjazz) [inline pontoon/settings/base.py:756]: Nit: Please remove the extra blank line.

**Review by mathjazz — COMMENTED** [review]
Thanks for the patch! Nice work!

I'd like the logging settings to remain exactly the same as they are ATM.

All the changes (logging to a file, logging format) should be optional and controlled via environment variables.

**Comment by RafaelJohn9** [comment]
> Thanks for the patch! Nice work!
> 
> I'd like the logging settings to remain exactly the same as they are ATM.
> 
> All the changes (logging to a file, logging format) should be optional and controlled via environment variables.

sure thing I'll update that

**Comment by RafaelJohn9** [comment]
hey @mathjazz , I have updated the PR, sorry for the late update.

**Comment by mathjazz** [comment]
Thanks for the update!

Logging is apparently still more verbose than on `main`, e.g. I see the following message in the logs:

```
[server] [DEBUG:django.db.backends] 2024-07-03 10:24:58,130 (0.000) SELECT "base_project"."id", "base_project"."total_strings", "base_project"."approved_strings", "base_project"."pretranslated_strings", "base_project"."strings_with_errors", "base_project"."strings_with_warnings", "base_project"."unreviewed_strings", "base_project"."name", "base_project"."slug", "base_project"."data_source", "base_project"."can_be_requested", "base_project"."configuration_file", "base_project"."disabled", "base_project"."date_created", "base_project"."date_disabled", "base_project"."sync_disabled", "base_project"."system_project", "base_project"."visibility", "base_project"."langpack_url", "base_project"."info", "base_project"."deadline", "base_project"."priority", "base_project"."contact_id", "base_project"."admin_notes", "base_project"."latest_translation_id", "base_project"."tags_enabled", "base_project"."pretranslation_enabled" FROM "base_project" WHERE "base_project"."id" IN (1) ORDER BY "base_project"."id" ASC; args=(1,); alias=default
```

Could we split this patch into two pieces:
1. Fix #3192, i.e. only add ability to optionally log to a FILE. 
2. File a separate bug to enable more verbose logging, and hash out the motiviation for it in the issue first.

**Comment by RafaelJohn9** [comment]
> Thanks for the update!
> 
> Logging is apparently still more verbose than on `main`, e.g. I see the following message in the logs:
> 
> ```
> [server] [DEBUG:django.db.backends] 2024-07-03 10:24:58,130 (0.000) SELECT "base_project"."id", "base_project"."total_strings", "base_project"."approved_strings", "base_project"."pretranslated_strings", "base_project"."strings_with_errors", "base_project"."strings_with_warnings", "base_project"."unreviewed_strings", "base_project"."name", "base_project"."slug", "base_project"."data_source", "base_project"."can_be_requested", "base_project"."configuration_file", "base_project"."disabled", "base_project"."date_created", "base_project"."date_disabled", "base_project"."sync_disabled", "base_project"."system_project", "base_project"."visibility", "base_project"."langpack_url", "base_project"."info", "base_project"."deadline", "base_project"."priority", "base_project"."contact_id", "base_project"."admin_notes", "base_project"."latest_translation_id", "base_project"."tags_enabled", "base_project"."pretranslation_enabled" FROM "base_project" WHERE "base_project"."id" IN (1) ORDER BY "base_project"."id" ASC; args=(1,); alias=default
> ```
> 
> Could we split this patch into two pieces? Namely:
> 
>     1. In this patch, fix [Ability to log in a file #3192](https://github.com/mozilla/pontoon/issues/3192), i.e. only add ability to optionally log to a file.
> 
>     2. File a separate bug to enable more verbose logging, and hash out the motivation for it in the issue first.

Sure thing. Sorry about the more verbose logging, I'll  remove the more verbose logging and focus on the task at hand

**Comment by RafaelJohn9** [comment]
@mathjazz  I have removed the simple, logging option, I have looked at the verbose logging options and I noticed it's currently the same as yours.

**Comment by RafaelJohn9** [comment]
hello @mathjazz  am really interested in this project, this PR is taking almost a month now 😅, if you don't mind could you give one of your social platforms you are active in so that we could talk more about how you want me to approach this issue, and hence a smooth workflow for future issues and PRs

File: pontoon/settings/base.py [hunk 2]
```diff
@@ -753,22 +753,69 @@ def _default_from_email():
...
+}
+
+# Define logging configuration
 LOGGING = {
     "version": 1,
     "disable_existing_loggers": False,
-    "handlers": {"console": {"class": "logging.StreamHandler"}},
+    "handlers": {
+        "console": console_handler,
+    },
```
Comment (mathjazz) [inline pontoon/settings/base.py:792]: Please restore this change. It doesn't seem needed to fix #3260. Then we can also remove `console_handler` above.

File: pontoon/settings/base.py [hunk 3]
```diff
@@ -753,22 +753,69 @@ def _default_from_email():
...
     "formatters": {
-        "verbose": {"format": "[%(levelname)s:%(name)s] %(asctime)s %(message)s"},
+        "verbose": {
+            "format": "[%(levelname)s:%(name)s] %(asctime)s %(message)s",
+        },
```
Comment (mathjazz) [inline pontoon/settings/base.py:796]: Please restore this change, it's just a formatting change.

File: pontoon/settings/base.py [hunk 4]
```diff
@@ -753,22 +753,69 @@ def _default_from_email():
...
     },
     "loggers": {
-        "django": {"handlers": ["console"]},
+        "django": {
+            "handlers": ["console"],
+            "level": os.environ.get("DJANGO_LOG_LEVEL", "DEBUG" if DEBUG else "INFO"),
+            "propagate": True,
+        },
```
Comment (mathjazz) [inline pontoon/settings/base.py:803]: Please restore this change. It doesn't seem needed to fix https://github.com/mozilla/pontoon/pull/3260.

File: pontoon/settings/base.py [hunk 5]
```diff
@@ -753,22 +753,69 @@ def _default_from_email():
...
         "pontoon": {
             "handlers": ["console"],
             "level": os.environ.get("DJANGO_LOG_LEVEL", "DEBUG" if DEBUG else "INFO"),
+            "propagate": True,
```
Comment (mathjazz) [inline pontoon/settings/base.py:807]: `propagate` defaults to `True`:
https://docs.djangoproject.com/en/5.0/howto/logging/#using-logger-hierarchies-and-propagation

So what's the motivation for this change?

**Comment by mathjazz** [comment]
Hey @RafaelJohn9, thanks for your contribution and patience!

See the README for contact information (I'm mathjazz on chat.mozilla.org and other channels):
https://github.com/mozilla/pontoon?tab=readme-ov-file#contributing-to-pontoon

**Comment by RafaelJohn9** [comment]
> Hey @RafaelJohn9, thanks for your contribution and patience!
> 
> See the README for contact information (I'm mathjazz on chat.mozilla.org and other channels):
> https://github.com/mozilla/pontoon?tab=readme-ov-file#contributing-to-pontoon
> 
> As mentioned [earlier](https://github.com/mozilla/pontoon/pull/3260#issuecomment-2205749711), #3192 is only about adding the ability to log to a file. There's still a few changes in this PR that are not needed for that. 
> 
> As soon as this is addressed, we can ship this. 

I'll contact you tomorrow if you are okay with it, to talk more about this

**Comment by RafaelJohn9** [comment]
I have corrected the changes that you reviewed above.

**Comment by mathjazz** [comment]
Thanks for the update.

Functionality looks good, please fix the linting error and we're done!

**Comment by RafaelJohn9** [comment]
done, 👍
Learned a lot in this PR,
thanks @mathjazz  for your patience, 🤝 means a lot.

**Comment by mathjazz** [comment]
@RafaelJohn9 Could you please document the newly added environment variable:
https://github.com/mozilla/pontoon/blob/main/docs/admin/deployment.rst#environment-variables

Sorry that I missed that earlier!

**Comment by RafaelJohn9** [comment]
@mathjazz  done, lemme know if there is anything else required

**Review by mathjazz — APPROVED** [review]
Thanks for updating the docs. I've pushed a minimal text style change directly.

Nice work with the patch, let's merge!
